framework/JobHandler.py

```diff
@@ -626,19 +552,34 @@ def addRuns(self):
           ## out as soon as that happens so we don't hog the lock.
           if len(self.__queue) > 0:
             item = self.__queue.popleft()
-            if isinstance(item, Runners.ExternalRunner):
-              command = item.command
-              command = command.replace("%INDEX%",str(i))
-              command = command.replace("%INDEX1%",str(i+1))
-              command = command.replace("%CURRENT_ID%",str(self.__nextId))
-              command = command.replace("%CURRENT_ID1%",str(self.__nextId+1))
-              command = command.replace("%SCRIPT_DIR%",self.runInfoDict['ScriptDir'])
-              command = command.replace("%FRAMEWORK_DIR%",self.runInfoDict['FrameworkDir'])
-              command = command.replace("%WORKING_DIR%",item.getWorkingDir())
-              command = command.replace("%BASE_WORKING_DIR%",self.runInfoDict['WorkingDir'])
-              command = command.replace("%METHOD%",os.environ.get("METHOD","opt"))
-              command = command.replace("%NUM_CPUS%",str(self.runInfoDict['NumThreads']))
-              item.command = command
+
+            ## Okay, this is a little tricky, but hang with me here. Whenever
+            ## a code model is run, we need to replace some of its command
+            ## parameters. The way we do this is by looking at the job instance
+            ## and checking if the first argument (the self in
+            ## self.evaluateSample) is an instance of Code, if so, then we need
+            ## to replace the execution command. Is this fragile? Possibly. We may
+            ## want to revisit this on the next iteration of this code.
+            if len(item.args) > 0 and isinstance(item.args[0], Models.Code):
+              kwargs = {}
+              kwargs['INDEX'] = str(i)
+              kwargs['INDEX1'] = str(i+i)
+              kwargs['CURRENT_ID'] = str(self.__nextId)
+              kwargs['CURRENT_ID1'] = str(self.__nextId+1)
+              kwargs['SCRIPT_DIR'] = self.runInfoDict['ScriptDir']
+              kwargs['FRAMEWORK_DIR'] = self.runInfoDict['FrameworkDir']
```

Comment:
Hm, the code above in Model.py has another problem:
```
abspath = os.path.abspath(str(self.executable))
```
will work badly for executables that are on the path.  As in:
```
>>> os.path.abspath("ls")
'/home/joshcogliati/ls'
```
It may not be very common, but the NCSU cluster had different locations for some of the python  libraries on the head node and the cluster nodes.